### PR TITLE
fix: non-nullable check for block variable and removed ! in layout_ 

### LINKED ISSUE #7523: Check for undefined in vertical flyout layout function
### Check for duplicates

- [X] I have searched for similar issues before opening a new one.

### Description

In [`layout_`](https://github.com/google/blockly/blob/osd/core/flyout_vertical.ts#L227) in `flyout_vertical.ts` we access the block with `const block = item.block`. The result is nullable, so all later uses of `block` have to have a [non-null assertion](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#non-null-assertion-operator-postfix-)

To fix:
- Check whether `block` is `undefined` after setting `const block = item.block`.
- `continue` if `block` doesn't exist.
- Remove `!` from after the remaining uses of `block` within the for loop.

Related to https://github.com/google/blockly/issues/7522

To get set up for this issue:
- Run `npm install` from the root of the repository.
- Run `npm run build` from the root of the repository.
- Run `npm run start` from the root of the repository to launch the test playground.

### PR DESCRIPTION
**Pull request by rashmi29**
## The basics

- [x] I [validated my changes](https://developers.google.com/blockly/guides/contribute/core#making_and_verifying_a_change)

### Resolves

Fixes #7523 

### Proposed Changes

- Check whether block is undefined after setting const block = item.block.
- continue if block doesn't exist.
- Remove ! from after the remaining uses of block within the for loop.

### Reason for Changes

Non nullable check was not present in code. That might break the code when variable has no value assigned.

### Test Coverage

I did npm run and tested all the buttons before and after and checked the entire functionality after my change.

### Documentation

No documentation required

### CODE REVIEW

**Comment by rachel-fenichel** [comment]
Thanks for grabbing this issue!

Next steps:
- Run `npm run format` to fix some formatting issues and commit
- Sign the CLA, following the instructions from the bot above

I'll keep an eye on this PR and approve workflow runs as needed.

File: core/flyout_vertical.ts [hunk 1]
```diff
@@ -233,29 +233,32 @@ export class VerticalFlyout extends Flyout {
     for (let i = 0, item; (item = contents[i]); i++) {
       if (item.type === 'block') {
         const block = item.block;
-        const allBlocks = block!.getDescendants(false);
+        if(block == undefined || block == null) {
```
Comment (rachel-fenichel) [inline core/flyout_vertical.ts:236]: This can be just `if (block) {`

**Review by rachel-fenichel — CHANGES_REQUESTED** [review]

File: core/flyout_vertical.ts [hunk 2]
```diff
@@ -233,29 +233,32 @@ export class VerticalFlyout extends Flyout {
     for (let i = 0, item; (item = contents[i]); i++) {
       if (item.type === 'block') {
         const block = item.block;
-        const allBlocks = block!.getDescendants(false);
+        if (block == undefined || block == null) {
```
Comment (rachanavishwanath) [inline core/flyout_vertical.ts:236]: please squash your commits

Comment (rachanavishwanath) [inline core/flyout_vertical.ts:236]: This check is redundant, block == null checks both null and undefined. We would both checks with strict equality i.e. block === undefined || block === null

Comment (rachel-fenichel) [reply core/flyout_vertical.ts:236]: In this case `if (block)` will do fine, because we aren't trying to distinguish between the two cases.

Comment (rashmi29) [reply core/flyout_vertical.ts]: If I just add if (block) than it shows "'block' is possibly 'undefined'." error. So, I added condition if (block == null) as suggested by rachna.

**Comment by rachel-fenichel** [comment]
Hi @rashmi29,

I'm signing off for the weekend but I will take a look again on Monday. Thanks again for contributing, and have a great Grace Hopper if you're attending!

**Comment by humera811** [comment]
nit: In the description section under `Resolves` you can add a link to the issue
```
  Resolves 
            fixes #7523 : Check for undefined in vertical flyout layout function 
```

File: core/flyout_vertical.ts [hunk 3]
```diff
@@ -233,29 +233,32 @@ export class VerticalFlyout extends Flyout {
...
+        if (block == null) {
+          continue;
+        }
+        const allBlocks = block.getDescendants(false);
         for (let j = 0, child; (child = allBlocks[j]); j++) {
           // Mark blocks as being inside a flyout.  This is used to detect and
           // prevent the closure of the flyout if the user right-clicks on such
           // a block.
           child.isInFlyout = true;
         }
```
Comment (humera811) [inline core/flyout_vertical.ts:245]: Suggestion:
```
if (block) {
          const allBlocks = block.getDescendants(false);
          for (let j = 0, child; (child = allBlocks[j]); j++) {
            // Mark blocks as being inside a flyout.  This is used to detect and
            // prevent the closure of the flyout if the user right-clicks on such
            // a block.
            child.isInFlyout = true;
          }
```
or we could also do 
```
if (!block) continue;
const allBlocks = block.getDescendants(false);
      for (let j = 0, child; (child = allBlocks[j]); j++) {
        // Mark blocks as being inside a flyout. This is used to detect and
        // prevent the closure of the flyout if the user right-clicks on such
        // a block.
        child.isInFlyout = true;
      }
```

**Review by maribethb — APPROVED** [review]
Looks great! Thank you so much for the PR!

**Pull request merged**